### ISSUE TITLE
[java]A false positive about ImmutableField and @Setter

### ISSUE THREAD

**Issue by LynnBroe** [pmd/pmd#4527]
**Affects PMD Version:7.0.0-rc1** 

**Rule: ImmutableField**

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_design.html#immutablefield>

**Description:**
The first code sample uses the annotation `@Setter` which can add the `setter` method of the field `foobar1`. So the analysis results of the two cases should be equivalent because these two cases are equivalent. However, PMD reports a warning in the first sample and no warnings in the second one. PMD should not report a warning in `Case1` like the `Case2`.

**Code Sample demonstrating the issue:**
**Case1**
```java
import lombok.Setter;
class MyClass1 {
    @Setter
    private String[] foobar1 = new String[0];   //  report a warning
    public final String[] call1() {
        return foobar1;
    }
}
```
**Case2**
```java
class MyClass2 {
    private String[] foobar1 = new String[0];   //  report no warning
    public final String[] call1() {
        return foobar1;
    }
    public void setFoobar1(final String[] foobar1) {
        this.foobar1 = foobar1;
    }
}
```

**Expected outcome:**

PMD reports a violation at line 4 in the first case, but that's wrong. That's a false positive.

**Comment by anastasiia-koba** [comment]
I guess it's a duplicate of #4254

**Comment by oowekyala** [comment]
Thanks for noticing @anastasiia-koba
